gpu/gpu.go

```diff
@@ -133,31 +131,14 @@ func getCPUMem() (memInfo, error) {
 func CheckVRAM() (int64, error) {
 	gpuInfo := GetGPUInfo()
 	if gpuInfo.FreeMemory > 0 && (gpuInfo.Library == "cuda" || gpuInfo.Library == "rocm") {
-		return int64(gpuInfo.FreeMemory), nil
-	}
-	return 0, fmt.Errorf("no GPU detected") // TODO - better handling of CPU based memory determiniation
-}
+		// allocate 384MiB for llama.cpp overhead (outside of model)
+		overhead := uint64(384 * 1024 * 1024)
+		if gpuInfo.FreeMemory <= overhead {
+			return 0, nil
+		}
```

Comment:
I'm a little torn.  You put this here because it seems to be GPU specific (doesn't affect mac), but logically it feels to me like it makes more sense in llm.go:New where we're doing all the memory calculations.